### ISSUE TITLE
BadageDrawable attached to the wrong menu item after making the item visible

### ISSUE THREAD

**Issue by Amrjyniat** [material-components/material-components#110]
I have two menu items on the toolbar, the first item in the menu layout is hidden by default and becomes visible under some conditions, now I attached the BadageDrawable with the second menu item and it works fine, no problem.
The problem is the BadageDrawable is attached to the first menu item when becomes visible which isn't the intended item. then return attached the BadageDrawable with the second item after hiding the first item.

https://user-images.githubusercontent.com/61949013/191682188-9fa29ebd-1771-4dd1-a958-c5de18d0009c.mp4

### Reproduction steps
 - Create a menu layout with two items and hide the first item.
 - Attach the BD to the second item.
 - Show the first item on some conditions.

### Version number
 - 1.7.0-beta01

### Operating system and device
Windows 10, tested on Pixel API 27

**Comment by mobilekosmos** [comment]
Yeah, this is reproducible in other scenarios as well, sometimes you even get multiple badges on over the other, you can see while debugging that it is a viewGroup with multiple badges.
